FinSearchAPI/ShopwareProcess.php

```diff
@@ -45,30 +49,33 @@ class ShopwareProcess {
 
 	/**
 	 * @param int $start
-	 * @param int $length
+	 * @param int $count
 	 *
 	 * @return xmlInformation
 	 * @throws \Exception
 	 */
-	public function getAllProductsAsXmlArray( $start = 0, $length = 0 ) {
+	public function getAllProductsAsXmlArray( $start = 0, $count = 0 ) {
 		$response = new xmlInformation();
 
 		$this->customerRepository = Shopware()->Container()->get( 'models' )->getRepository( Customer::class );
-
-		// Get all articles from selected shop
-		/** @var array $allArticles */
-		$allArticles = $this->shop->getCategory()->getAllArticles();
-
-		$response->total = count($allArticles);
-
-		if ( $length > 0 ) {
-			$allArticles = array_slice( $allArticles, $start, $length );
+		$this->articleRepository  = Shopware()->Container()->get( 'models' )->getRepository( Article::class );
+		$this->orderRepository    = Shopware()->Container()->get( 'models' )->getRepository( Order::class );
+
+		if ( $count > 0 ) {
+			$articlesQuery = $this->articleRepository->createQueryBuilder( 'articles' )
+			                                       ->leftJoin( 'articles.details', 'details' )
+			                                       ->select( 'articles' )
+			                                       ->setMaxResults( $count )
+			                                       ->setFirstResult( $start );
+			/** @var array $allArticles */
+			$allArticles = $articlesQuery->getQuery()->execute();
 		}
-
+		else{
+			/** @var array $allArticles */
+			$allArticles = $this->shop->getCategory()->getAllArticles();
+		}
+		$response->total = count(  $this->shop->getCategory()->getAllArticles() );
```

Comment:
Wird so leider nicht funktionieren, weil wir wieder in das Speicherproblem laufen